pkg/virtual/framework/forwardingregistry/rest.go

```diff
@@ -66,35 +94,49 @@ func NewStorage(ctx context.Context, resource schema.GroupVersionResource, apiEx
 		// TODO: what do we do on Destroy()?
 	}
 
-	delegate := DefaultDynamicDelegatedStoreFuncs(
+	store := DefaultDynamicDelegatedStoreFuncs(
 		factory, listFactory, destroyer,
 		strategy, tableConvertor,
 		resource, apiExportIdentityHash, categories,
 		dynamicClusterClient, []string{}, *patchConflictRetryBackoff, ctx.Done(),
 	)
-	store := wrapper(resource.GroupResource(), delegate)
+	if wrapper != nil {
```

Comment:
You have an inherent `nil` check in the implementation of `Decorate()` - do we need it here?